File: Data Structures & Algorithms/duplicate-integer/submission-1.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        int n = nums.length;
       HashSet<Integer>st = new HashSet<>();
       for(int num : nums) {
        st.add(num);// add all elements into the hasset 
       }
       return st.size() != n;
    }
}

File: Data Structures & Algorithms/rotate-array/submission-1.cpp
class Solution {
public:
    void rotate(vector<int>& nums, int k) {
       int n = nums.size();
       int index = 0;
       vector<int>v(n);
        k = k % n; // k > n
       for(int i=n-k;i<n;i++){
        v[index++] = nums[i];
       } 
       // index = 4 ;
       for(int i=0; i < n-k; i++){
         v[index++] = nums[i];
       }
       for(int i=0;i<v.size();i++){nums[i] = v[i];}
       for(int num : nums) {cout<<num<<" ";}
    }
};